From e8a751d0f4b635891d838fb76d7da260b34c7b43 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Thu, 16 Aug 2018 14:34:40 +0800
Subject: [PATCH 4491/5242] MLK-18794-3 usb: cdns3: change the definition of
 PowerState at XECP_PM_PMCSR

commit  21c267669874471b5f94315704a2993a61d5d803 from
https://source.codeaurora.org/external/imx/linux-imx.git

When the USB port goes to suspend, PowerState should set to "D1"
(the D2 and D3hot are not supported now); PowerState should set to
"D0" when the USB port goes to resume.

Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/cdns3/cdns3-nxp-reg-def.h |    4 +++-
 drivers/usb/cdns3/core.c              |    6 ++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/usb/cdns3/cdns3-nxp-reg-def.h b/drivers/usb/cdns3/cdns3-nxp-reg-def.h
index 8866322..4ac7300 100644
--- a/drivers/usb/cdns3/cdns3-nxp-reg-def.h
+++ b/drivers/usb/cdns3/cdns3-nxp-reg-def.h
@@ -168,5 +168,7 @@
 #define CFG_RXDET_P3_EN		(1 << 15)
 
 /* XECP_PM_PMCSR */
-#define PS_D0			(1 << 0)
+#define PS_MASK			(3 << 0)
+#define PS_D0			0
+#define PS_D1			(1 << 0)
 #endif /* __DRIVERS_USB_CDNS3_NXP_H */
diff --git a/drivers/usb/cdns3/core.c b/drivers/usb/cdns3/core.c
index 4726718..2ba35c0 100644
--- a/drivers/usb/cdns3/core.c
+++ b/drivers/usb/cdns3/core.c
@@ -714,7 +714,8 @@ static void cdns3_enter_suspend(struct cdns3 *cdns, bool suspend, bool wakeup)
 
 		/* SW request low power when all usb ports allow to it ??? */
 		value = readl(xhci_regs + XECP_PM_PMCSR);
-		value |= PS_D0;
+		value &= ~PS_MASK;
+		value |= PS_D1;
 		writel(value, xhci_regs + XECP_PM_PMCSR);
 
 		/* mdctrl_clk_sel */
@@ -774,7 +775,8 @@ static void cdns3_enter_suspend(struct cdns3 *cdns, bool suspend, bool wakeup)
 
 		/* SW request D0 */
 		value = readl(xhci_regs + XECP_PM_PMCSR);
-		value &= ~PS_D0;
+		value &= ~PS_MASK;
+		value |= PS_D0;
 		writel(value, xhci_regs + XECP_PM_PMCSR);
 
 		/* clr CFG_RXDET_P3_EN */
-- 
1.7.9.5

